core/GlobalState.cc

```diff
@@ -504,6 +504,10 @@ void GlobalState::initEmpty() {
                  .untypedArg(Names::arg2())
                  .buildWithResult(Types::rangeOfUntyped());
 
+    // Synthesize <Magic>,<regex-backref>() => T.nilable(String)
```

Comment:
typo: `,` → `.`

and actually: what do you think about making this take the backref kind as an argument?

```
<Magic>.<regex-backref>(:"$&")
```

etc.

You should be able to do this pretty easily by just calling something like

```cpp
MK::Symbol(backref->loc, backref->name)
```